py/fastspecfit/io.py

```diff
@@ -1368,6 +1368,14 @@ def read_stacked(self, stackfiles, firsttarget=0, ntargets=None,
             
             ivar = fitsio.read(stackfile, 'IVAR')
             ivar = ivar[fitindx, :]
+            
+            # Check if the file contains a resolution matrix, if it does not
+            # then use an identity matrix
+            try:
+                res = fitsio.read(stackfile, 'RES')
+                res = res[fitindx, :, :]
+            except:
+                res = np.stack([np.ones((1, npix)) for i in range(nobj)], axis=0) # Hack!
```

Comment:
Could this `for` loop ever become slow in the limit of large `nobj`? Isn't there a pure-numpy way to generate the array we need here?

Reply:
Changed now. This was a really dumb looking line 😂